torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,429 @@
+        """Merge the modalities that are downloaded and extracted in two parts.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the filename of the modality
+        """
+        fname_p1 = modality + "_part1"
+        fname_p2 = modality + "_part2"
+        source_folder = os.path.join(self.root, fname_p1)
+        destination_folder = os.path.join(self.root, fname_p2)
+        for file_name in os.listdir(source_folder):
+            source = os.path.join(source_folder, file_name)
+            destination = os.path.join(destination_folder, file_name)
+            if os.path.isfile(source):
+                shutil.move(source, destination)
+
+        shutil.rmtree(source_folder)
+        dest_split = os.path.split(destination_folder)
+        if len(modality.split("_")) == 3:
+            rename_dest = os.path.join(dest_split[0], "s2_temporal_subset")
+        if len(modality.split("_")) == 2:
+            rename_dest = os.path.join(
+                dest_split[0],
+                dest_split[1].split("_")[0] + "_" + dest_split[1].split("_")[1],
+            )
+        if "_" not in modality:
+            rename_dest = os.path.join(dest_split[0], dest_split[1].split("_")[0])
+        os.rename(destination_folder, rename_dest)
+
+    def _convert_to_color(
+        self, arr_2d: Tensor, cmap: dict[int, tuple[int, int, int]]
+    ) -> "np.typing.NDArray[np.uint8]":
+        """Numeric labels to RGB-color encoding.
+
+        Args:
+            arr_2d: 2D array to be colorized
+            cmap: colormap to use when mapping the labels
+
+        Returns:
+            3D colored image
+        """
+        arr_3d = np.zeros((arr_2d.shape[0], arr_2d.shape[1], 3), dtype=np.uint8)
+
+        for c, i in cmap.items():
+            m = arr_2d == c
+            arr_3d[m] = i
+        return arr_3d
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample image-mask pair returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        modality_channels = {"viirs": 1, "esa_wc": 1, "s1": 2}
+
+        # Assign a number of channels to all 's2' modalities
+        for modality in self.modality:
+            if modality.startswith("s2"):
+                modality_channels[modality] = 10
+
+        start_idx = 0
+        split_images = {}
+
+        for modality in self.modality:
+            end_idx = start_idx + modality_channels[modality]  # Start + n of channels
+            split_images[modality] = sample["image"][start_idx:end_idx, :, :]  # Slicing
+            start_idx = end_idx  # Update the iterator
+
```

Comment:
Above this line could be a function that outputs the `split_images`. I have no preference